packages/strapi-plugin-documentation/services/Documentation.js

```diff
@@ -1284,12 +1288,67 @@ module.exports = {
     if (verb === 'get' && controllerMethod === 'find') {
       // parametersOptions corresponds to this section
       // of the documentation https://strapi.io/documentation/guides/filters.html
-      return [...params, ...parametersOptions];
-    }
 
+      const apiName = this.extractAPIName(routeObject);
+      if (this.hasAPIName(apiName)) {
+        return this.combineMembersAndParameters(apiName, params);
+      }
+    }
     return params;
   },
 
+  /**
+   * Extract the api route from the endpoint.
+   * @param endPoint
+   * @return the API route.
+   */
+  extractAPIName: routeObject => {
```

Comment:
I don't think you need all of this as in the caller of the generateVerbParameters you alredy avec the apiName or pluginName and the route.handler to extract the controllerName and method. 

You should be able to call generateVerbParameters directly with the model or at  least `{ pluginName, apiName }` to avoid parsing the endpoint at all

Reply:
OK, I will to improve this tomorrow.

Reply:
Instead of parsing the apiName variable I am using the variable passed by the calling method `generateApiDocumentation`. When the `generatePluginDocumentation` method calls `generateVerbParameters` I am passing null though. I hope this is the right approach. The commit eliminated by doing that some lines of code. Also no need for using the pluralize plugin.